neuron-explainer/neuron_explainer/explanations/simulator.py

```diff
@@ -590,51 +594,183 @@ def _format_record_for_logprob_free_simulation(
+            return zero_prediction
+        predicted_activations = []
+        # check that there is a token and activation value
+        # no need to double check the token matches exactly
+        for i, activation in enumerate(activations):
+            if "token" not in activation:
+                logger.error("The key 'token' is not in activation:\n%s\nCompletion:%s", activation, json.dumps(completion))
+                predicted_activations.append(0)
+                continue
+            if "activation" not in activation:
+                logger.error("The key 'activation' is not in activation:\n%s\nCompletion:%s", activation, json.dumps(completion))
+                predicted_activations.append(0)
+                continue
+            # Ensure activation value is between 0-10 inclusive
+            try:
+                predicted_activation_float = float(activation["activation"])
+                if predicted_activation_float < 0 or predicted_activation_float > MAX_NORMALIZED_ACTIVATION:
+                    logger.error("activation value out of range: %s\nCompletion:%s", predicted_activation_float, json.dumps(completion))
+                    predicted_activations.append(0)
+                else:
+                    predicted_activations.append(predicted_activation_float)
+            except ValueError:
+                logger.error("activation value invalid: %s\nCompletion:%s", activation["activation"], json.dumps(completion))
+                predicted_activations.append(0)
+            except TypeError:
+                logger.error("activation value incorrect type: %s\nCompletion:%s", activation["activation"], json.dumps(completion))
+                predicted_activations.append(0)
+        logger.debug("predicted activations: %s", predicted_activations)
+        return predicted_activations
+    
+    except json.JSONDecodeError:
+        logger.error("Failed to parse completion JSON:\n%s\nExpected Tokens:\n%s", completion, tokens)
+        return zero_prediction
 
 def _parse_no_logprobs_completion(
     completion: str,
     tokens: Sequence[str],
-) -> Sequence[int]:
+) -> Sequence[float]:
     """
     Parse a completion into a list of simulated activations. If the model did not faithfully
     reproduce the token sequence, return a list of 0s. If the model's activation for a token
-    is not an integer betwee 0 and 10, substitute 0.
+    is not a number between 0 and 10 (inclusive), substitute 0.
 
     Args:
         completion: completion from the API
         tokens: list of tokens as strings in the sequence where the neuron is being simulated
     """
+
+    logger.debug("for tokens:\n%s", tokens)
+    logger.debug("received completion:\n%s", completion)
+
     zero_prediction = [0] * len(tokens)
-    token_lines = completion.strip("\n").split("༗\n")
+    # FIX: Strip the last ༗\n, otherwise all last activations are invalid
```

Comment:
good catch! let's remove the `FIX`, I don't think it makes sense out of context of the PR :)